Aztec/Classes/TextKit/TextView.swift

```diff
@@ -909,23 +909,42 @@ open class TextView: UITextView {
 
         let originalText = attributedText.attributedSubstring(from: range)
         let attributedTitle = NSAttributedString(string: title)
-        let finalRange = NSRange(location: range.location, length: attributedTitle.length)        
+        let finalRange = NSRange(location: range.location, length: attributedTitle.length)
 
         undoManager?.registerUndo(withTarget: self, handler: { [weak self] target in
             self?.undoTextReplacement(of: originalText, finalRange: finalRange)
         })
 
         let formatter = LinkFormatter()
-        formatter.attributeValue = url        
+        formatter.attributeValue = url
         let attributes = formatter.apply(to: typingAttributes)
+        let underLineStr = addUnderlineInLink(title: title, attributes: attributes)
```

Comment:
@arisun instead of adding the underline attributes here you should do it inside the LinkFormatter class above.

The formatter classes are our central place to apply attributes and they are used in all place that the link attributes are needed. 

The way you implemented the underline attribute is only applied when the user create a link using the UI not when a link is read directly from the HTML.